addons/point_of_sale/static/src/app/screens/product_screen/control_buttons/control_buttons.js

```diff
@@ -48,18 +49,23 @@ export class ControlButtons extends Component {
                 item: fiscalPos,
             });
         }
-        this.dialog.add(SelectionPopup, {
-            title: _t("Select Fiscal Position"),
+
+        const selectedFiscalPosition = await makeAwaitable(this.dialog, SelectionPopup, {
             list: fiscalPosList,
-            getPayload: (selectedFiscalPosition) => {
-                this.currentOrder.set_fiscal_position(selectedFiscalPosition);
-                // IMPROVEMENT: The following is the old implementation and I believe
-                // there could be a better way of doing it.
-                for (const line of this.currentOrder.orderlines) {
-                    line.set_quantity(line.quantity);
-                }
-            },
+            title: _t("Please register the voucher number"),
         });
+
+        if (!selectedFiscalPosition) {
+            this.currentOrder.set_fiscal_position(false);
+            return;
+        }
```

Comment:
```suggestion
        if (!selectedFiscalPosition) {
            return;
        }
```
Is this the logic when the user press Escape when the selection popup is shown? Resetting isn't a good UX. We should keep the current fiscal position in the order.